src/local-models/Llama3/README.md

```diff
@@ -0,0 +1,182 @@
+- **Enhanced Privacy**: Data remains on your local device, offering a secure environment for sensitive information.
+- **Customization**: Local models allow for greater flexibility to tweak and optimize the models as per your needs3.
+
+## Prerequisites
+
+To run Phi-3 or Llama3 model on your local machine, the following prerequisites are required:
+
+- [Install ollama](https://ollama.com/download). Ollama can be installed in macOS, Linux or Windows. 
+- Download and run a LLM. 
+    [Ollama library](https://ollama.com/library) provides different types of open source models. To install Llama3 run the command:
+
+    ```bash
+    ollama run llama3
+    ```
+- The command `ollama run phi3` will install Phi-3.
+
+- The command `ollama list` will show the local installed models:
+
+    ![Image Description](img/10ollamarunllama3.png)
+
+***Tip**: To see the ollama service journal in realtime, you can run the command:*
+
+```bash
+journalctl -u ollama -f
+```
+
+
+## Quick Start, using Llama3 model as an example
+
+The llama3 model is locally running, we can now run some prompts using C# and Semantic Kernel.
+
+### Step 1: Run the model
+
+The file `src/local-models/Llama3/Program.cs` shows an example on how to use Semantic Kernel to use the local model. When adding the OpenAI Chat Completion service to the kernel, we must specify the modelId, endpoint and apikey (usually an empty string).
+
+### Chat Completion
+```csharp
+using Microsoft.SemanticKernel;
+
+// Create kernel with a custom http address
+var builder = Kernel.CreateBuilder();
+builder.AddOpenAIChatCompletion(
+    modelId: "llama3",
+    endpoint: new Uri("http://localhost:11434"),
+    apiKey: "apikey");
+var kernel = builder.Build();
+
+var prompt = "Write a joke about kittens. Use emojis.";
+var response = await kernel.InvokePromptAsync(prompt);
+Console.WriteLine(response.GetValue<string>());
+
+```
+#### Output 
+```
+Why did the kitten join a band?
+
+Because it wanted to be the purr-cussionist!
+```
+
+### Chat Completion with Phi3
+
+Using the same endpoint, we can now select Phi3 as the model to be used.
+
+```csharp
+using Microsoft.SemanticKernel;
+
+// Create kernel with a custom http address
+var builder = Kernel.CreateBuilder();
+builder.AddOpenAIChatCompletion(
+    modelId: "phi3",
+    endpoint: new Uri("http://localhost:11434"),
+    apiKey: "apikey");
+var kernel = builder.Build();
+
+var prompt = "Write a joke about kittens. Use emojis.";
+var response = await kernel.InvokePromptAsync(prompt);
+Console.WriteLine(response.GetValue<string>());
+
+```
+#### Output 
+```
+Why did the scaredy-kitten refuse to walk through the forest at night?
+
+Because it heard there were "nocturnal" predators out for its dinner! But really, they're just as sleepy during the day like we all are. Kitties need their beauty sleep too, right after playtime with a ball of yarn! ???
+
+Remember, humor about animals should always be respectful and harmless.
+```
```

Comment:
```suggestion
### Chat Completion

#### Phi-3

```csharp
using Microsoft.SemanticKernel;

// Create kernel with a custom http address
var builder = Kernel.CreateBuilder();
builder.AddOpenAIChatCompletion(
    modelId: "phi3",
    endpoint: new Uri("http://localhost:11434"),
    apiKey: "apikey");
var kernel = builder.Build();

var prompt = "Write a joke about kittens. Use emojis.";
var response = await kernel.InvokePromptAsync(prompt);
Console.WriteLine(response.GetValue<string>());

```

##### Output 
```
Why did the scaredy-kitten refuse to walk through the forest at night?

Because it heard there were "nocturnal" predators out for its dinner! But really, they're just as sleepy during the day like we all are. Kitties need their beauty sleep too, right after playtime with a ball of yarn! ???

Remember, humor about animals should always be respectful and harmless.
```

#### Llama 3

```csharp
using Microsoft.SemanticKernel;

// Create kernel with a custom http address
var builder = Kernel.CreateBuilder();
builder.AddOpenAIChatCompletion(
    modelId: "llama3",
    endpoint: new Uri("http://localhost:11434"),
    apiKey: "apikey");
var kernel = builder.Build();

var prompt = "Write a joke about kittens. Use emojis.";
var response = await kernel.InvokePromptAsync(prompt);
Console.WriteLine(response.GetValue<string>());

```
##### Output 
```
Why did the kitten join a band?

Because it wanted to be the purr-cussionist!
```
```